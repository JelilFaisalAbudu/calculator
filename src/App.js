import React from 'react';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        Hello, React!
      </header>
    </div>
  );
}

export default App;
