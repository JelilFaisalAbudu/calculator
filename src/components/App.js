import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import appStyles from '../styles/app.module.css';
import calculate from '../logic/calculate';

const App = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = buttonName => {
    const result = calculate({ total, next, operation }, buttonName);

    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  };

  const renderResult = `${total}${operation}${next}`.replace(/null/g, '');

  return (
    <div className={appStyles.app}>
      <p className={appStyles.heading}>Let&apos;s do some math!</p>
      <div
        className="calculator-board"
        style={{ width: 700 }}
      >
        <Display result={renderResult || undefined} />
        <ButtonPanel clickHandler={handleClick} />
      </div>
    </div>
  );
};

export default App;
