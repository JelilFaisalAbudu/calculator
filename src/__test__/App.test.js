import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import App from '../components/App';

describe('App', () => {
  test('renders app component', () => {
    render(<App />);
    const textNode = screen.getByText('Let\'s do some math!');
    expect(textNode).toBeInTheDocument();
  });

  test('renders the app component correctly', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
