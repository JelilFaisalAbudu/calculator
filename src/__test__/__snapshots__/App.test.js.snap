// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app component correctly 1`] = `
<div
  className="app"
>
  <p
    className="heading"
  >
    Let's do some math!
  </p>
  <div
    className="calculator-board"
    style={
      Object {
        "width": 700,
      }
    }
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="button-panel"
    >
      <div
        className="button-group"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e3913e",
              "width": "25%",
            }
          }
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="button-group"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e3913e",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="button-group"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e3913e",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="button-group"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e3913e",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="button-group"
      >
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e0e0e0",
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#e3913e",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
